docs/api/en/core/Object3D.html

```diff
@@ -58,6 +58,13 @@ <h3>[property:Boolean frustumCulled]</h3>
 		<h3>[property:Integer id]</h3>
 		<p>readonly – Unique number for this object instance.</p>
 
+		<h3>[property:Boolean inFrustum]</h3>
+		<p>
+			If frustumCulled is set, this is set if the object is within the frustum of the camera in the last render pass. Default is *false*.<br /><br />
+
+			This is not checked if .visible is false or the object fails the layer test.
```

Comment:
Potentially, otherwise you have to do the frustum checks when not needed hence the notes.

Maybe a different name? .wasRendered which would be more accurate and probably fulfil much the same purpose. Then it could take into account the material visibility checks.

.isVisible was suggested in the original proposal but was thought to be potentially misleading with occlusion not being considered.

Reply:
Sidenote: You would not have inconsistency issues when working with callbacks. If certain conditions are not true (e.g.`visible` is set to `false`), the callbacks just don't fire. You don't have the problem that a property represents an obsolete value.

Reply:
Yes, we already have onBeforeRender() as well, so another callback seems redundant, but we don't have the negative callback onNotRendered(). 

One thing I considered is recording the last frame number an object is rendered in.  
I already use this to get simple LRU tracking.

Reply:
Um...I was referring to something like `onBecameVisibleI()` and `onBecameInvisible()` like mentioned here: https://github.com/mrdoob/three.js/issues/17271#issuecomment-522303305

Reply:
@Mugen87  Did you make a comment and then delete it, regarding onBecameVisible() etc? I think implementing them would require too much state to be maintained per camera for useful core support.  

You could derive an equivalent to onBecameVisible() via onBeforeRender() in application code, and use a post render check to work out which objects became invisible in that pass so the current set of callbacks is probably enough?

Reply:
I've deleted the comment because I was not sure how to write it without creating confusion. Realized yesterday that this flag approach is somewhat inconsistent to the rest of the API. `three.js` does normally not set such read-only flags. An approach with methods/callbacks might be better in that regard.